### ISSUE TITLE
What version of az cli does this task use?

### ISSUE THREAD

**Issue by 13daysaweek** [MicrosoftDocs/azure-devops-docs#10660]
It would be great of this document calls out what version of az cli the task actually uses.  Is it pinned to a specific version?  Does it always use the latest?  If not, when does the az cli version increment?  Only when the version of the task increments?  Of course I can run the task and call `az --version` but it would be great to document what the expected behavior for az cli version is with this task as customers can have dependencies on specific versions of az cli.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8de3304e-98f9-bac0-3ccc-c65f68a89e87
* Version Independent ID: 8fb2f6a9-545d-dff1-ff86-974694a85054
* Content: [Azure CLI task - Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-cli?view=azure-devops)
* Content Source: [docs/pipelines/tasks/deploy/azure-cli.md](https://github.com/MicrosoftDocs/azure-devops-docs/blob/master/docs/pipelines/tasks/deploy/azure-cli.md)
* Product: **devops**
* Technology: **devops-cicd-tasks**
* GitHub Login: @ushan
* Microsoft Alias: **UshaN**

**Comment by MS-alsulli** [comment]
Hey @13daysaweek , member of the DevOps support team here. Came across this issue. This question is a big "it depends". 

TL;DR: CLI version depends on a number of factors, but ultimately comes down to what is on the machine you're running the task on (whether it's ms-hosted agents or a private agent).

It depends on if you're using Microsoft hosted agents or private agents. If you're using Microsoft hosted, it depends on even more variables, such as what agent image you're using (Windows, Linux, mac), as well as the geographic region where your Azure DevOps organization is located. Azure DevOps does a rolling deployment, meaning some customers get updates before others. Will touch on this a bit more in detail below.

The actual task itself doesn't dictate the version of the Azure CLI. It all depends on what version of the CLI is on the image you're using. Let's say as an **example** you're using the Microsoft Hosted Windows 2019 image. Details on the software and tools installed on the image can be found here: https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md#cli-tools (this is the anchor to the CLI version, but you can scroll through and see the other tools). Currently the Azure CLI version on the Windows 2019 image is 2.25.0 (at time of writing). These images update roughly every week. Let's say a new version of Azure CLI was released tomorrow (**hypothetical**). It would be roughly a week or two, depending on a number of factors, until the hosted images were updated with the latest version of the Azure CLI. As I stated before, the MS hosted images roll out in a rolling deployment schedule. If the Azure CLI was upgraded  on the hosted images then began deploying, it may take two weeks or so on top of that before all customers see it.

If you're using a private agent on your own infrastructure, then the Azure CLI version only changes when you upgrade / downgrade that tooling.

Hope this provides some clarity for you!

**Comment by ramiMSFT** [comment]
@13daysaweek  Thank you for your feedback!
@MS-alsulli Thank you for the detailed response!
